src/app/portlets/content-types/fields/dot-content-type-fields-variables/dot-content-type-fields-variables-table-row/dot-content-type-fields-variables-table-row.component.html

```diff
@@ -0,0 +1,63 @@
+        <p-cellEditor>
+            <ng-template pTemplate="input">
+                <input pInputText
+                    type="text"
+                    [placeholder]="messages['contenttypes.field.variables.key_input.placeholder']"
+                    (focus)="editFieldInit()"
+                    (keyup)="editFieldInit()"
+                    (keydown.escape)="onCancel($event)"
+                    (keydown.enter)="onPressEnter($event)"
+                    [(ngModel)]="fieldVariable.key">
+            </ng-template>
+            <ng-template pTemplate="output">
+                <span *ngIf="fieldVariable.key">
+                    {{ fieldVariable.key }}
+                </span>
+                <span class="empty-placeholder" *ngIf="!fieldVariable.key">
+                    {{ messages['contenttypes.field.variables.key_input.placeholder'] }}
+                </span>
+            </ng-template>
+        </p-cellEditor>
+    </td>
+    <td #valueCell [pEditableColumn]="fieldVariable.value" [pEditableColumnField]="'value'">
+        <p-cellEditor>
+            <ng-template pTemplate="input">
+                <input pInputText
+                    type="text"
+                    [placeholder]="messages['contenttypes.field.variables.value_input.placeholder']"
+                    (focus)="editFieldInit()"
+                    (keyup)="editFieldInit()"
+                    (keydown.escape)="onCancel($event)"
+                    (keydown.enter)="onPressEnter()"
+                    [(ngModel)]="fieldVariable.value">
+            </ng-template>
+            <ng-template pTemplate="output">
+                <span *ngIf="fieldVariable.value">
+                    {{ fieldVariable.value }}
+                </span>
+                <span class="empty-placeholder" *ngIf="!fieldVariable.value">
+                    {{ messages['contenttypes.field.variables.value_input.placeholder'] }}
+                </span>
+            </ng-template>
+        </p-cellEditor>
+    </td>
+    <td class="content-type-fields__variables-actions">
+        <div id="content-type-fields__variables-actions-edit" *ngIf="showEditMenu">
+            <button pButton secondary tiny
+                [label]="messages['contenttypes.action.cancel']"
+                (click)="onCancel($event)"></button>
+            <button #saveButton
+                pButton tiny
+                class="content-type-fields__variables-actions-edit-save"
+                [label]="messages['contenttypes.action.save']"
+                [disabled]="saveDisabled"
+                (click)="saveVariable()"></button>
+        </div>
+        <div id="content-type-fields__variables-actions-main" *ngIf="!showEditMenu">
+            <dot-icon-button icon="delete_outline" (click)="delete.emit(variableIndex)"></dot-icon-button>
+            <dot-icon-button icon="edit" (click)="focusKeyInput(true)"></dot-icon-button>
+        </div>
+    </td>
+</tr>
```

Comment:
maybe we can split this component, maybe we can get out each td content into a new component

Reply:
IMO not needed, since it's too simple